contrib/babelfishpg_tsql/sql/sys_views.sql

```diff
@@ -1075,23 +1136,24 @@ select cast(t.typname as text) as name
   , 0 as is_assembly_type
   , 0 as default_object_id
   , 0 as rule_object_id
-  , case when is_tbl_type then 1 else 0 end as is_table_type
+  , case when tt.typrelid is not null then 1 else 0 end as is_table_type
 from pg_type t
-inner join pg_namespace s on s.oid = t.typnamespace
 join sys.schemas sch on t.typnamespace = sch.schema_id
+left join sys.type_info_internal ti on t.typname = ti.pg_type_name
 left join pg_collation c on c.oid = t.typcollation
-, sys.translate_pg_type_to_tsql(t.oid) AS tsql_type_name
+left join sys.table_types_internal tt on t.typrelid = tt.typrelid
 , sys.translate_pg_type_to_tsql(t.typbasetype) AS tsql_base_type_name
```

Comment:
Is there a reason to keep this?

Reply:
Yes, this call relates to user defined types which are not handled in sys.type_info_internal, therefore we need this call.
In the future, someone may take a look if the user defined types (which can be added and removed at any time) can be translated in a view efficiently.